packages/core/test/requestCollection.spec.ts

```diff
@@ -169,184 +177,177 @@ describe('fetch tracker', () => {
 describe('xhr tracker', () => {
   let originalOpen: typeof XMLHttpRequest.prototype.open
   let originalSend: typeof XMLHttpRequest.prototype.send
-  let requestObservable: Observable<RequestDetails>
-  let notifySpy: jasmine.Spy
+  let requestStartObservable: Observable<RequestStartEvent>
+  let requestCompleteObservable: Observable<RequestCompleteEvent>
+  let startSpy: jasmine.Spy
+  let completeSpy: jasmine.Spy
 
   beforeEach(() => {
     originalOpen = XMLHttpRequest.prototype.open
     originalSend = XMLHttpRequest.prototype.send
-    requestObservable = new Observable<RequestDetails>()
-    notifySpy = spyOn(requestObservable, 'notify').and.callThrough()
-    trackXhr(requestObservable)
+    requestStartObservable = new Observable()
+    requestCompleteObservable = new Observable()
+    startSpy = spyOn(requestStartObservable, 'notify').and.callThrough()
+    completeSpy = spyOn(requestCompleteObservable, 'notify').and.callThrough()
+    trackXhr([requestStartObservable, requestCompleteObservable])
   })
 
   afterEach(() => {
     XMLHttpRequest.prototype.open = originalOpen
     XMLHttpRequest.prototype.send = originalSend
   })
 
-  function xhrSpec(setup: (xhr: XMLHttpRequest) => void, expectations: (xhr: XMLHttpRequest) => void, done: DoneFn) {
+  function xhrSpec({
+    done,
+    setup,
+    expectedMethod,
+    expectedResponse,
+    expectedStatus,
+    expectedURL,
+    expectXHR,
+  }: {
+    done: DoneFn
+    setup: (xhr: XMLHttpRequest) => void
+    expectedMethod: string | jasmine.Any
+    expectedStatus: number | jasmine.Any
+    expectedURL: string
+    expectedResponse?: string | jasmine.Any
+    expectXHR?: (xhr: XMLHttpRequest) => void
+  }) {
     const xhr = new XMLHttpRequest()
     xhr.addEventListener('loadend', () => {
       setTimeout(() => {
-        expect(notifySpy).toHaveBeenCalledWith(
-          jasmine.objectContaining({
-            duration: jasmine.any(Number),
-            startTime: jasmine.any(Number),
-            traceId: undefined,
-            type: 'xhr',
-          })
-        )
-        expectations(xhr)
+        const completeEvent = (completeSpy.calls.allArgs() as RequestCompleteEvent[][])
+          .map((args) => args[0])
+          // IE10/11 doesn't have .find()
+          .filter((d) => includes(d.url, expectedURL))[0]
```

Comment:
we probably don't need to add that comment everywhere we use some of the utils polyfills

Reply:
This is for the `find()` method, not `includes()`. Maybe I could add a util `find()` function?

Reply:
oh ok, yes adding find coud be handy